File: 5nomer.kt
import java.util.*

fun proverka(num: Long): Int { // Функция поиска чётных чисел
    var a = Math.abs(num)
    var res = 0

    if (a == 0L) {
        return 1
    }

    while (a > 0) {
        val promej = (a % 10).toInt()
        if (promej % 2 == 0) {
            res++
        }
        a /= 10
    }
    return res
}

fun vivod(num: Long, res: Int) { // Функция вывода
    println("Число $num имеет $res чётных цифр")
}

fun vvodChisla(firstvvod: Boolean): Long { // Функция ввода
    val scanner = Scanner(System.`in`)
    if (firstvvod) {
        print("Введите число (для завершения введите любое нечисленное значение): ")
    }
    return if (scanner.hasNextLong()) {
        scanner.nextLong()
    } else {
        -1 // Вернем -1 как признак окончания ввода
    }
}

fun main() {
    var firstvvod = true
    val scanner = Scanner(System.`in`)
    
    while (true) {
        val num = vvodChisla(firstvvod)
        firstvvod = false
        
        if (num == -1L) break
        
        val res = proverka(num)
        vivod(num, res)
    }
}

File: 5nomer.cpp
#include<iostream>
using namespace std;

int proverka(long long num) { // Функия поиска чётных чисел
 int a = abs(num);
 int res = 0;
 
 if(a == 0){
  return 1;
 }
 
 while (a > 0){
  int promej = a % 10;
  if(promej % 2 == 0){
   res++;
  }
  a /= 10;
 }
 return res;
}

void vivod(long long num, int res){ // Функция вывода
 cout << "Число " << num << " имеет " << res << " чётных чисел" << endl;
}

long long vvodChisla(bool firstvvod){ // Функция ввода
 long long num;
 if(firstvvod){
  cout << "Введите число (для завершения введите любое нечисленное значение): ";
 }
 cin >> num;
 return num;
 
}

int main(){
 bool firstvvod = true;
 while(cin){
  long long num = vvodChisla(firstvvod);
  firstvvod = false;
  if(cin.fail()) break;
  int res = proverka(num);
  vivod(num, res);
 }
 return 0;
}
